src/Appwrite/Migration/Version/V22.php

```diff
@@ -346,14 +349,29 @@ private function migrateDocument(Document $document): Document
                     ->setAttribute('deploymentResourceType', $document->getAttribute('deploymentResourceType', $deploymentResourceType))
                     ->setAttribute('search', \implode(' ', [$document->getId(), $document->getAttribute('domain', '')]));
 
-                if ($deploymentResourceType === 'function') {
-                    $project = $this->dbForProject->getDocument('projects', $document->getAttribute('projectId'));
+                $project = $this->dbForProject->getDocument('projects', $document->getAttribute('projectId'));
 
-                    if ($project->isEmpty()) {
-                        Console::warning("Project \"{$document->getAttribute('projectId')}\" not found for rule \"{$document->getId()}\"");
-                        break;
-                    }
+                if ($project->isEmpty()) {
+                    Console::warning("Project \"{$document->getAttribute('projectId')}\" not found for rule \"{$document->getId()}\"");
+                    $document->setAttribute('region', System::getEnv('_APP_DEFAULT_REGION'));
+                    break;
+                }
+
+                $document->setAttribute('region', $project->getAttribute('region', System::getEnv('_APP_DEFAULT_REGION')));
+
+                $domain = $document->getAttribute('domain', '');
+                $functionsDomain = System::getEnv('_APP_DOMAIN_FUNCTIONS', '');
+                $sitesDomain = System::getEnv('_APP_DOMAIN_SITES', '');
+                $owner = '';
+                if (
+                    (!empty($functionsDomain) && \str_ends_with($domain, $functionsDomain)) ||
+                    (!empty($sitesDomain) && \str_ends_with($domain, $sitesDomain))
+                ) {
+                    $owner = 'Appwrite';
+                }
+                $document->setAttribute('owner', $document->getAttribute('owner', $owner));
```

Comment:
```suggestion
                if (
                    (!empty($functionsDomain) && \str_ends_with($domain, $functionsDomain)) ||
                    (!empty($sitesDomain) && \str_ends_with($domain, $sitesDomain))
                ) {
                    $document->setAttribute('owner', 'Appwrite');
                }
```

should be the same right?

Reply:
We want to keep whatever value is in the attribute if one is set already. Otherwise, set it to Appwrite if that if resolves to true.

Reply:
How about:

```php
      $domain = $document->getAttribute('domain', '');
      $functionsDomain = System::getEnv('_APP_DOMAIN_FUNCTIONS', '');
      $sitesDomain = System::getEnv('_APP_DOMAIN_SITES', '');
      $owner = $document->getAttribute('owner', '');
      if (
          empty($owner) &&
          (!empty($functionsDomain) && \str_ends_with($domain, $functionsDomain)) ||
          (!empty($sitesDomain) && \str_ends_with($domain, $sitesDomain))
      ) {
          $document->setAttribute('owner', 'Appwrite');
      }
```